### PR TITLE
Snap Runtime Management refactor

### LINKED ISSUE #208: Refactor proposal: Make an object to represent a single snap
The `SnapsController` (and execution environment services) has a series of maps that each point to the snaps that it manages.

This means when we want to do things like unregister event listeners, we end up going down each of these maps, and unsetting them, and in general it means that data that really is per-snap is sprawled across the code that applies to the entire snap controller.

I think there is a cleanup/refactor we could do where we create a `Snap` class that represents a single initialized snap. This would allow us to keep initialization and teardown steps very neatly isolated to the object being handled, and makes it easier to make further changes to our model (for example, if we decide to initialize a snap per persona).

This isn't urgent, but it's a refactor that I've consistently found myself wanting.

### PR DESCRIPTION
**Pull request by FrederikBolding**
Combines the several maps used to keep track of information in `SnapsController` into a single map.

Closes #208

### CODE REVIEW

File: packages/controllers/src/snaps/SnapController.ts [hunk 1]
```diff
@@ -396,18 +415,12 @@ export class SnapController extends BaseController<
    * A {@link Map} of Snap IDs and the Unix timestamp of the most recent RPC
    * request received by the Snap.
    */
```
Comment (rekmarks) [inline packages/controllers/src/snaps/SnapController.ts:417]: This docstring should be deleted.

File: packages/controllers/src/snaps/SnapController.ts [hunk 2]
```diff
@@ -108,6 +108,25 @@ export type Snap = {
   version: string;
 };
 
+export interface SnapRuntimeData {
```
Comment (rekmarks) [inline packages/controllers/src/snaps/SnapController.ts:111]: This could use a docstring explaining why it exists and why we don't keep it in state.

**Review by rekmarks — CHANGES_REQUESTED** [review]
Looks good! Just a couple of comment-related requests.

File: packages/controllers/src/snaps/SnapController.ts (hunk 1 above)
Comment (FrederikBolding) [reply packages/controllers/src/snaps/SnapController.ts]: Done!

File: packages/controllers/src/snaps/SnapController.ts (hunk 2 above)
Comment (FrederikBolding) [reply packages/controllers/src/snaps/SnapController.ts:115]: Done!

**Review by rekmarks — APPROVED** [review]
LGTM!

**Pull request merged**